File: ca341/euler-168-number-rotations.lua/euler-168-number-rotations.lua
function GetFiveSum(n)
    local sum = 0
    for i=11, tonumber(n) do
        local rotation = GetRotation(i)
        if IsDivisible(i, rotation) then
            sum = sum + i
        end
    end
    return string.sub(tostring(sum), -5)
end

function GetRotation(n)
    local r = n % 10
    local m = math.floor(n / 10)
    return math.floor(m + (r * (10 ^ m)))
end

function IsDivisible(n, rotation)
    return rotation % n == 0
end

print(GetFiveSum(arg[1]))

File: 3/ca341/euler-168-number-rotations.rb/euler-168-number-rotations.rb
class NumberRotations
    def initialize(n)
        @upper_limit = n
        @sum_of_rotations = get_five_sum(@upper_limit)
    end

    def get_five_sum(n)
        sum = 0
        for i in 11..n.to_i do
            rotation = get_rotation(i)
            if is_divisible(i, rotation) then
                sum = sum + i
            end
        end
        return sum.to_s[-5..-1]
    end
    
    def get_rotation(n)
        r = n % 10
        m = (n / 10).floor
        return (m + (r * (10 ** m))).floor
    end
    
    def is_divisible(n, rotation)
        return rotation % n == 0
    end

    def to_s()
        to_str
    end

    def to_str()
        @sum_of_rotations
    end
end

nr = NumberRotations.new(ARGV[0])
puts nr.to_s